pyfar/signals/deterministic.py

```diff
@@ -256,6 +393,491 @@ def exponential_sweep_time(n_samples, frequency_range, n_fade_out=90,
+        try to increase `n_samples`, `start_margin`, `stop_margin`, `fade_in`
+        or `fade_out`, or provide a more smooth magnitude spectrum.
+
+    Parameters
+    ----------
+    n_samples : int
+        The length of the sweep in samples.
+    magnitude_spectrum : Signal
+        The magnitude spectrum as a Signal object. Must be at least `n_samples`
+        long and is zero-padded to `n_samples` otherwise.
+    start_margin : int, float
+        The time in samples, at which the sweep starts. The start margin is
+        required because the frequency domain sweep synthesis has pre-ringing
+        in the time domain.
+    stop_margin : int, float
+        Time in samples, at which the sweep stops. This is relative to
+        `n_samples`, e.g., a stop margin of 100 samples means that the sweep
+        ends at sample ``n_samples-100``. This is required, because the
+        frequency domain sweep synthesis has post-ringing in the time domain.
+    fade_in : int, optional
+        Duration of a squared sine fade-in in samples. The fade starts at the
+        first sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``None`` does not apply
+        a fade-in.
+    fade_out : int, optional
+        Duration of a squared cosine fade-out in samples. The fade ends at the
+        last sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``None`` does not apply
+        a fade-out.
+    sampling_rate : int, optional
+        The sampling rate in Hz. The default is ``44100``.
+    group_delay : boolean, optional
+        Return the analytical group delay of the sweep. This can be used to
+        compute the times at which distortion products appear. The default is
+        ``False``.
+
+    Returns
+    -------
+    sweep : Signal
+        The sweep signal. The Signal is in the time domain, has a maximum
+        absolute amplitude of 1 and the ``none`` FFT normalization
+        (see :py:func:`~pyfar.dsp.fft.normalization`).
+    group_delay_sweep : FrequencyData
+        The analytical group delay of the sweep in seconds as a single sided
+        spectrum. Only returned if `group_delay` is ``True``.
+
+    References
+    ----------
+    .. [#] S. Müller, P. Massarani. 'Transfer Function Measurement with Sweeps.
+           Directors Cut Including Previously Unreleased Material and some
+           Corrections. J. Audio Eng. Soc. 2001, 49 (6), 443-471.
+
+    Examples
+    --------
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> magnitude = pf.dsp.filter.low_shelve(
+        ...     pf.signals.impulse(2**16), 500, 20, 2)
+        >>> magnitude = pf.dsp.filter.butterworth(magnitude, 8, 50, 'highpass')
+        >>> sweep = pf.signals.magnitude_spectrum_weighted_sweep(
+        ...     2**16, magnitude, 5000, 1000)
+        >>> pf.plot.time_freq(sweep)
```

Comment:
How about replacing this with an example that might be a bit more relevant, like an interpolated octave spectrum with target signal level? Similar to what you might be interested in when targeting a specific SNR in room acoustic measurements?

Reply:
Id like to keep it simple, here - especially while our interpolation function still overshoots :(

Reply:
Ah yes, maybe we can make an wxample Notebook at some point